src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -403,6 +470,7 @@ private void TrackNullableStateForAssignment(BoundNode node, Symbol assignmentTa
                         {
                             ReportStaticNullCheckingDiagnostics(ErrorCode.WRN_NullReferenceAssignment, (value ?? node).Syntax);
                         }
+                        this.State[slot] = valueIsNotNull;
```

Comment:
>this.State[slot] = valueIsNotNull; [](start = 24, length = 34)

Changing state for this code path feels inconsistent with behavior of InheritNullableStateOfFieldOrProperty. I would like to suggest to not make any changes in this area in context of this PR. Let's focus on tracking members

Reply:
For reference types, the default value is `null` meaning unassigned, so it's necessary to update the state on assignment so that the state for members is populated (lazily) with declared nullability when those members are accessed (`y.F` below for instance):
```
class C
{
    object? F;
    static void M(C x)
    {
        C y = x;
        object z = y.F;
    }
}
```

Reply:
Removed this assignment. `null` state is no longer treated as unassigned.